sds_data_manager/stacks/ialirt_ingest_lambda_stack.py

```diff
@@ -0,0 +1,159 @@
+from aws_cdk import aws_lambda_python_alpha as lambda_alpha_
+from constructs import Construct
+
+
+class IalirtIngestLambda(Stack):
+    """Stack for ialirt ingest lambda."""
+
+    def __init__(
+        self,
+        scope: Construct,
+        construct_id: str,
+        env: cdk.Environment,
+        ialirt_bucket: aws_s3.Bucket,
+        **kwargs,
+    ) -> None:
+        """IalirtIngestLambda Stack.
+
+        Parameters
+        ----------
+        scope : Construct
+            Parent construct.
+        construct_id : str
+            A unique string identifier for this construct.
+        env : obj
+            Environment.
+        ialirt_bucket : aws_s3.Bucket
+            The data bucket.
+        kwargs : dict
+            Keyword arguments.
+
+        """
+        super().__init__(scope, construct_id, env=env, **kwargs)
+
+        # Create DynamoDB Table
+        self.packet_data_table = self.create_dynamodb_table()
+
+        # Create Lambda Function
+        self.ialirt_ingest_lambda = self.create_lambda_function(
+            ialirt_bucket, self.packet_data_table
+        )
+
+        # Create Event Rule
+        self.create_event_rule(ialirt_bucket, self.ialirt_ingest_lambda)
+
+    def create_dynamodb_table(self) -> aws_dynamodb.Table:
+        """Create and return the DynamoDB table."""
+        table = ddb.Table(
+            self,
+            "PacketDataTable",
+            table_name="imap-packetdata-table",
+            # Change to RemovalPolicy.RETAIN to keep the table after stack deletion.
+            # TODO: change to RETAIN in production.
+            removal_policy=RemovalPolicy.DESTROY,
+            # Restore data to any point in time within the last 35 days.
+            # TODO: change to True in production.
+            point_in_time_recovery=False,
+            # Partition key (PK) = spacecraft time ugps.
+            partition_key=ddb.Attribute(
+                name="reset_number#met",
+                type=ddb.AttributeType.STRING,
+            ),
+            # Define the read and write capacity units.
+            # TODO: change to provisioned capacity mode in production.
+            billing_mode=ddb.BillingMode.PAY_PER_REQUEST,  # On-Demand capacity mode.
+        )
+        return table
+
+    def create_lambda_function(
+        self, ialirt_bucket: aws_s3.Bucket, packet_data_table: aws_dynamodb.Table
+    ) -> lambda_alpha_.PythonFunction:
+        """Create and return the Lambda function."""
+        lambda_role = iam.Role(
+            self,
+            "IalirtIngestLambdaRole",
+            assumed_by=iam.ServicePrincipal("lambda.amazonaws.com"),
+            managed_policies=[
+                iam.ManagedPolicy.from_aws_managed_policy_name(
+                    "service-role/AWSLambdaBasicExecutionRole"
+                )
+            ],
+        )
+
+        lambda_role.add_managed_policy(
+            iam.ManagedPolicy.from_aws_managed_policy_name("AmazonDynamoDBFullAccess")
+        )
```

Comment:
Move this into the managed policies above.
```suggestion
                )
            iam.ManagedPolicy.from_aws_managed_policy_name("AmazonDynamoDBFullAccess"),
            ],
        )
```